examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -142,20 +142,31 @@ class _PestoDemoState extends State<PestoDemo> {
           new DrawerItem(
             child: new Text('Home'),
             selected: !config.showFavorites,
-            onPressed: () => Navigator.pushNamed(context, PestoDemo.routeName)
+            onPressed: () {
+              Navigator.popUntil(context, (Route<dynamic> route) {
+                return !route.willHandlePopInternally
+                  && route is ModalRoute && route.settings.name == '/pesto';
```

Comment:
Maybe we should provide a currying function that returns this predicate...

So this becomes:
```
Navigator.popUntil(context, ModalRoute.matches('/pesto'));
```
```
// in ModalRoute:
static RoutePredicate matches(String routeName) {
  return (Route<dynamic> route) {
    return !route.willHandlePopInternally && route is ModalRoute && route.settings.name == routeName;
  }
}
```